src/mono/mono/mini/mini.h

```diff
@@ -316,11 +316,13 @@ enum {
 #define MONO_IS_ZERO(ins) (((ins)->opcode == OP_VZERO) || ((ins)->opcode == OP_XZERO))
 
 #ifdef TARGET_ARM64
-// SIMD is only supported on arm64 when using the LLVM backend. When not using
-// the LLVM backend, treat SIMD datatypes as regular value types.
-#define MONO_CLASS_IS_SIMD(cfg, klass) ( ((cfg)->opt & MONO_OPT_SIMD) && ( COMPILE_LLVM (cfg) ) && m_class_is_simd_type (klass) )
+/*
+ * SIMD is only supported on arm64 when using the LLVM backend. When not using
+ * the LLVM backend, treat SIMD datatypes as regular value types.
+ */
+#define MONO_CLASS_IS_SIMD(cfg, klass) (((cfg)->opt & MONO_OPT_SIMD) && COMPILE_LLVM (cfg) && m_class_is_simd_type (klass))
 #else
-#define MONO_CLASS_IS_SIMD(cfg, klass) (((cfg)->opt & MONO_OPT_SIMD) && m_class_is_simd_type (klass))
+#define MONO_CLASS_IS_SIMD(cfg, klass) (((cfg)->opt & MONO_OPT_SIMD) && m_class_is_simd_type (klass) && (COMPILE_LLVM (cfg) || mono_type_size (m_class_get_byval_arg (klass), NULL) == 16))
 #endif
```

Comment:
This means that non-arm64 only supports vector128 in non-llvm mode right ?

Reply:
Yeah. Maybe the condition should be made more specific, and MONO_CLASS_IS_SIMD should expand to a constant "false" expression on everything other than x86, amd64, and arm64.